### PR TITLE
KAFKA-17106:enable testFetchProgressWithMissingPartitionPosition

### PR DESCRIPTION
**Pull request by TaiJuWu**
*More detailed description of your change,
See discussion https://github.com/apache/kafka/pull/16541#discussion_r1671273572

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@TaiJuWu could you please rebase code to check the hangings ?

**Comment by TaiJuWu** [comment]
> @TaiJuWu could you please rebase code to check the hangings ?

Done.

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java [hunk 1]
```diff
@@ -850,24 +850,26 @@ public void verifyNoCoordinatorLookupForManualAssignmentWithOffsetCommit(GroupPr
...
 
         consumer = newConsumerNoAutoCommit(groupProtocol, time, client, subscription, metadata);
         consumer.assign(Arrays.asList(tp0, tp1));
         consumer.seekToEnd(singleton(tp0));
         consumer.seekToBeginning(singleton(tp1));
 
+        if (groupProtocol == GroupProtocol.CONSUMER) {
+            Node node = metadata.fetch().nodes().get(0);
+            client.prepareResponseFrom(FindCoordinatorResponse.prepareResponse(Errors.NONE, groupId, node), node);
+        }
```
Comment (FrankYang0529) [inline clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java:866]: May you move this block before `newConsumerNoAutoCommit`? Please check https://issues.apache.org/jira/browse/KAFKA-17129 for the reason.

tl;dr [MockClient#send](https://github.com/apache/kafka/blob/7495e70365ebeba4806fc4da23c2ca75fa191649/clients/src/test/java/org/apache/kafka/clients/MockClient.java#L215-L278) only match request when there is a `FutureResponse`. If we don't add it to `MockClient` first, `FindCoordinatorRequest` may not get a response and it will hang for long time.

**Review by FrankYang0529 — COMMENTED** [review]
@TaiJuWu thanks for the PR. Could you check the comment about a minor change? Thanks.

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java (hunk 1 above)
Comment (TaiJuWu) [reply clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java:866]: Thanks for your information. Update.

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks.

**Review by chia7712 — APPROVED** [review]
LGTM
